### DISCUSSION TITLE
[VSCode] Alternative: Repository configuration folders

### DISCUSSION THREAD

**Discussion by Jan-Kowalik-mulberryrisk** [devcontainers/community#51] [Q&A] [answered]
Can you provide more guidance on how to configure `"dev.containers.repositoryConfigurationPaths"` or at least troubleshoot it, please?
I have tried to follow the very minimal docs [here](https://code.visualstudio.com/docs/devcontainers/create-dev-container#_alternative-repository-configuration-folders) but it does not work and I can't figure out how to set it up to find the devcontainer file.

My dir structure is as follows:
```
workspace
├── my_project.code-workspace
├── repo-containers
│   └── github.com
│       └── MyOrg
│           └── my_project
│               ├── Dockerfile
│               └── devcontainer.json
└── src
    └── my_project
```

The `code-workspace` file points to `src/my_project` directory where code from `git@github.com:MyOrg/my_project.git` is checked out
How do I configure `dev.containers.repositoryConfigurationPaths` to find and use the `devcontainer.json` file?

Thanks in advance.

**Comment by Jan-Kowalik-mulberryrisk** [comment]
Is anyone here?

**Reply by Chuxel** [reply]
The `dev.containers.repositoryConfigurationPaths` capability is set up to pick up the configuration when you open a folder from the cloned repository. It is not intended to be workspace specific since you can use devcontainer.json file there directly.  So, typically it wouldn't be under "workspace" in your example above. But, with this file structure, it should be set to the full path of "workspace/repo-containers".   

If this is still not working, you raise an issue for the VS Code Dev Containers extension at https://github.com/microsoft/vscode-remote-release

**Reply by Jan-Kowalik-mulberryrisk** [reply]
`workspace` dir is not to mean a VSCode workspace it is just a parent folder of my work stuff. It could just as well be named `projects`, a bit unfortunate name, I know. I will try the path you advised and report back. Thank you.

Do you use that config option yourself?

**Reply by Jan-Kowalik-mulberryrisk** [reply]
@Chuxel, Are you able to tell me what it uses as a base directory for that relative path?
Are there any logs that I can see to know what it fails on?

**Reply by Jan-Kowalik-mulberryrisk** [reply]
anyone?
There is virtually no documentation for this feature

**Reply by Chuxel** [reply, marked as answer]
The path should be an absolute path on your filesystem (full path, not relative) - you're partly not getting responses here since best spot for the Dev Containers VS Code extension is at https://github.com/microsoft/vscode-remote-release

**Reply by Jan-Kowalik-mulberryrisk** [reply]
Thanks, I think I finally got it. It _clicked_ to me now that for it to work you need to open the folder in a container and not open the workspace in a container. Also I kept using workspace relative path.
I have tried doing that on a remote Linux/WSL machine too. That is too much of an _inception_ going on for the feature I guess, with a container env being started from remote machine env.

It would be appreciated if the docs where clearer and state this only works when opening folders and that the setting value must be an absolute system path.
Additionally for someone like me, that starting a remote container env from another remote env will not work either.


> you're partly not getting responses here since best spot for the Dev Containers VS Code extension is at https://github.com/microsoft/vscode-remote-release

They tell you that issue tracker is not for asking questions and send you to StackOverflow. My [post](https://stackoverflow.com/questions/76338546/how-do-i-configure-dev-containers-alternative-repository-configuration-folders-i) there has no responses neither.

**Comment by WellspringCS** [comment]
FWIW, I just experienced a fair bit of what @Jan-Kowalik-mulberryrisk describes. I agree that WSL doesn't work. I also agree that documentation of this feature is negligible. I also proved (only by experimentation, since documentation was not clear) that this feature only works (if it does at all, and for me that's a theoretical statement) for the actual .devcontainer folder, and not anything else you might want to link in.

My only success was to (after trying 4 flavors of folders in WSL and Windows both, guns-a-blazing) finally convince VSCode to find my *Windows* .devcontainer folder, and to suggest opening the project in a dev container. When I said yes, VS Code failed to load the json file, bombing out with some non-descript error that left me in the dark as to what it didn't like. 

I gave up.
